packages/flutter/lib/src/cupertino/action_sheet.dart

```diff
@@ -0,0 +1,1265 @@
+  /// reason for the alert.
+  ///
+  /// Typically a [Text] widget.
+  final Widget message;
+
+  /// The set of actions that are displayed for the user to select.
+  ///
+  /// Typically this is a list of [CupertinoActionSheetAction] widgets.
+  final List<Widget> actions;
+
+  /// A scroll controller that can be used to control the scrolling of the
+  /// [message] in the action sheet.
+  ///
+  /// This attribute is typically not needed, as alert messages should be
+  /// short.
+  final ScrollController messageScrollController;
+
+  /// A scroll controller that can be used to control the scrolling of the
+  /// [actions] in the action sheet.
+  ///
+  /// This attribute is typically not needed.
+  final ScrollController actionScrollController;
+
+  /// The optional cancel button that is grouped separately from the other
+  /// actions.
+  ///
+  /// Typically this is an [CupertinoActionSheetAction] widget.
+  final Widget cancelButton;
+
+  Widget _buildContent() {
+    final List<Widget> content = <Widget>[];
+    if (title != null || message != null) {
+      final Widget titleSection = new _CupertinoAlertContentSection(
+        title: title,
+        message: message,
+        scrollController: messageScrollController,
+      );
+      content.add(new Flexible(child: titleSection));
+    }
+
+    return new Container(
+      color: _kBackgroundColor,
+      child: new Column(
+        mainAxisSize: MainAxisSize.min,
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        children: content,
+      ),
+    );
+  }
+
+  Widget _buildActions() {
+    if (actions == null || actions.isEmpty) {
+      return new Container(
+        height: 0.0,
+      );
+    }
+    return new Container(
+      child: new _CupertinoAlertActionSection(
+        children: actions,
+        scrollController: actionScrollController,
+        hasCancelButton: cancelButton != null,
+      ),
+    );
+  }
+
+  Widget _buildCancelButton() {
+    final double cancelPadding = (actions != null || message != null || title != null)
+        ? _kCancelButtonPadding : 0.0;
+    return Padding(
+      padding: new EdgeInsets.only(top: cancelPadding),
+      child: new _CupertinoActionSheetCancelButton(
+        child: cancelButton,
+      ),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final List<Widget> children = <Widget>[
+      new Flexible(child: new ClipRRect(
+        borderRadius: new BorderRadius.circular(12.0),
+        child: new BackdropFilter(
+          filter: new ImageFilter.blur(sigmaX: _kBlurAmount, sigmaY: _kBlurAmount),
+          child: new Container(
+            decoration: _kAlertBlurOverlayDecoration,
+            child: new _CupertinoAlertRenderWidget(
+              contentSection: _buildContent(),
+              actionsSection: _buildActions(),
+            ),
+          ),
+        ),
+      ),
+      ),
+    ];
+
+    if (cancelButton != null) {
+      children.add(
+        _buildCancelButton(),
```

Comment:
Is there no blur behind the cancel button?

Reply:
No, from observation/experimentation I've found that the cancel button isn't transparent like the other buttons.

Reply:
Ah cool, thanks for testing